behaviors/arm/src/arm_component.cc

```diff
@@ -17,43 +17,66 @@
  */
 
 // Standard ROS includes
-#include <ros/ros.h>
-#include <nodelet/nodelet.h>
-#include <pluginlib/class_list_macros.h>
+#include <rclcpp/rclcpp.hpp>
 
 // Standard sensor messages
-#include <sensor_msgs/JointState.h>
+#include <sensor_msgs/msg/joint_state.hpp>
+namespace sensor_msgs {
+typedef msg::JointState JointState;
+}  // namespace sensor_msgs
 
 // FSW shared libraries
 #include <config_reader/config_reader.h>
 #include <ff_util/config_server.h>
-#include <ff_util/ff_nodelet.h>
+#include <ff_util/ff_component.h>
 #include <ff_util/ff_action.h>
 #include <ff_util/ff_fsm.h>
+#include <ff_util/ff_service.h>
 
 // FSW actions, services, messages
-#include <ff_msgs/ArmAction.h>
-#include <ff_msgs/ArmStateStamped.h>
-#include <ff_msgs/JointSampleStamped.h>
-#include <ff_msgs/SetState.h>
-#include <ff_hw_msgs/SetEnabled.h>
-#include <ff_hw_msgs/CalibrateGripper.h>
+#include <ff_msgs/action/arm.hpp>
+#include <ff_msgs/msg/arm_state.hpp>
+#include <ff_msgs/msg/arm_state_stamped.hpp>
+#include <ff_msgs/msg/joint_sample.hpp>
+#include <ff_msgs/msg/joint_sample_stamped.hpp>
+#include <ff_msgs/msg/arm_joint_state.hpp>
+#include <ff_msgs/msg/arm_gripper_state.hpp>
+#include <ff_msgs/srv/set_state.hpp>
+namespace ff_msgs {
+typedef action::Arm Arm;
+typedef msg::ArmState ArmState;
+typedef msg::ArmStateStamped ArmStateStamped;
+typedef msg::JointSample JointSample;
+typedef msg::JointSampleStamped JointSampleStamped;
+typedef msg::ArmJointState ArmJointState;
+typedef msg::ArmGripperState ArmGripperState;
+typedef srv::SetState SetState;
+}  // namespace ff_msgs
+
+#include <ff_hw_msgs/srv/set_enabled.hpp>
+#include <ff_hw_msgs/srv/calibrate_gripper.hpp>
+namespace ff_hw_msgs {
+typedef srv::SetEnabled SetEnabled;
+typedef srv::CalibrateGripper CalibrateGripper;
+}  // namespace ff_hw_msgs
 
 /**
  * \ingroup beh
  */
 namespace arm {
 
+FF_DEFINE_LOGGER("arm");
+
 // Different joint types
 enum JointType { PAN, TILT, GRIPPER };
 
 // Perching arm servos to enable / disable
-  enum ServoID {
-    PROXIMAL_SERVO,           // Proximal joint servo
-    DISTAL_SERVO,             // Distal joint servo
-    GRIPPER_SERVO,            // Gripper joint servo
-    ALL_SERVOS          // Proximal, Distal and Gripper servos
-  };
+enum ServoID {
+  PROXIMAL_SERVO,           // Proximal joint servo
+  DISTAL_SERVO,             // Distal joint servo
+  GRIPPER_SERVO,            // Gripper joint servo
+  ALL_SERVOS          // Proximal, Distal and Gripper servos
+};
 
 // Joint information, where HUMAN = SCALE * DRIVER + OFFSET
 struct JointInfo {
```

Comment:
Now that we aren't using tol (I'm pretty sure) except to set it but never look at it, I would just delete it from JointInfo to make things more clear.

Reply:
That sounds like a change to apply/test on `develop` first?

Reply:
I thought it only applies here due to changes in how the parameter server works, but correct me if I'm wrong Marina.

Reply:
In order to maintain functionality with the previous version the tol with GetTolerance() still only gets updated in certain flight modes, which maintains the need for the tol variable :)